core/handlers.py

```diff
@@ -32,27 +45,30 @@ def handle_validation_error(validation_error: ValidationError) -> Exception | tu
     }, 400
 
 
-def handle_exception(uncaught_exception: Exception) -> Exception | tuple[Response, int]:
+def handle_exception(uncaught_exception: Exception) -> Exception | tuple[dict, int]:
     # pass through HTTP errors
     if isinstance(uncaught_exception, HTTPException):
         return uncaught_exception
 
     # handle uncaught ingest errors
     if request.path == "/ingest":
-        failure_uuid = uuid.uuid4()
-        s3_object_name = FAILED_FILE_S3_NAME_FORMAT.format(failure_uuid, datetime.now().strftime(DATETIME_ISO_8601))
-        upload_file(file=g.excel_file, bucket=Config.AWS_S3_BUCKET_FAILED_FILES, object_name=s3_object_name)
-        current_app.logger.error(f"Uncaught ingest exception - failure_id={str(failure_uuid)}", exc_info=True)
-        return (
-            jsonify(
-                {
-                    "detail": "Uncaught ingest exception.",
-                    "id": failure_uuid,
-                    "status": 500,
-                    "title": "Internal Server Error",
-                }
-            ),
-            500,
-        )
+        return handle_ingest_error()
 
     return uncaught_exception
+
+
+def handle_ingest_error():
+    """Saves the file that fails ingest to S3 and returns a generic uncaught exception 500 error.
+
+    :return: A 500 error response
+    """
+    failure_uuid = uuid.uuid4()
+    s3_object_name = FAILED_FILE_S3_NAME_FORMAT.format(failure_uuid, datetime.now().strftime(DATETIME_ISO_8601))
+    upload_file(file=g.excel_file, bucket=Config.AWS_S3_BUCKET_FAILED_FILES, object_name=s3_object_name)
+    current_app.logger.error(f"Uncaught ingest exception - failure_id={str(failure_uuid)}", exc_info=True)
+    return {
+        "detail": "Uncaught ingest exception.",
```

Comment:
Can we return any more detail about the error in this 500 message? Can we pass {internal_failures} from the calling function and return it here. Might help when building a pipeline to see them here

Reply:
Good shout - I've added this so it returns as part of the response